backend/cms/models/site.py

```diff
@@ -22,14 +22,14 @@ class Site(models.Model):
         (ARCHIVED, 'Archived'),
     )
 
-    title = models.CharField(max_length=200)
-    name = models.URLField(max_length=60, unique=True)
+    name = models.CharField(max_length=200)
+    slug = models.CharField(max_length=200, unique=True)
     status = models.CharField(max_length=4, choices=STATUS)
-    language_tree = models.ForeignKey(LanguageTree, null=True, on_delete=models.SET_NULL)
+    languages = models.ManyToManyField(Language)
```

Comment:
Should we be able to enable / disable only a few languages of a Site? For example during the time in which a Site is working on a new language, it shouldn't be delivered to the API; what do you expect this `languages` field to look like in this situation? Should it include the new language?

Reply:
Also, since you're removing the tree structure: How do you save the information in which direction the pages should be translated? Afaik you said on the conference, that this should be remembered per page. Although the most common (or even the only) use case is that all pages of one site are translated in the same direction.

Reply:
tbh I didn't think of the case that they are working on a language which should not be published yet. The solution I think of is that we allow any language in the content creation, but deliver only the languages of this field in the api. If this isn't a good approach, we have to add another option as `api_languages` or so.

Concerning the translation direction:
As far as I understood, the XLIFF-files already contain the information, which language they're coming from. But you're right, in this case we'd need another field `translation_origin` or similar in the `PageTranslation` model. @sascha11110 @svenseeberg @Ponitac what do you think, would you prefer the language tree model?

Reply:
Having an option to disable the listing of a language sounds useful to me. On the other hand, this issue did not come up yet.

And yes, I agree with Timo that we should store the translation origin on a per page basis. The XLIFF contains information about the source language.

Reply:
The current workflow is as follows: When a new language is added, it is delivered via the API, but their translations are not public. This way, the language will appear in the WebApp, but the user can not change the language in the language switcher. Instead, the message "Translation not available" is shown when hovering over the language. I think this approach is also perfectly fine in the Django cms, so we don't need an additional field here.

Apart from that, I added `source_language` and `currently_in_translation` to the page translation model. @Schedulaar do you agree with this?

Reply:
> [...] I think this approach is also perfectly fine in the Django cms, so we don't need an additional field here.

Well, it's not exactly the same -- the language will be shown (and selectable) after selecting the city for example and then there would be no content in that site-language pair (unpleasant)...
Also these kind of things can only be introduced with a major model-change (like introducing a site-language-model and having a property `live: boolean` on that). That is true for any property a Site may want to set per language (an imaginable scenario imo). Changing a model once there's data in the DB is always a huge pain in the a** (if we want some kind of per-language-properties we need to do complex db-migrations), whereas adding a property to a model is very easy (if we chose now to add a language-site-model or similar). So we should be sure whether we should introduce sth like that now already (also balancing flexibility vs complexity in models).

> Apart from that, I added source_language [...] to the page translation

Well, I introduced the not-per-page-but-per-site-source-language(-tree) (:D) as a result of a discussion with @dkehne:
Some city had started translating their German pages to several destination pages (all sourced on the German language) manually on their own, but messed up with the translation sources (the sources weren't all set to German, but to mixed languages on mixed pages).
At some point they decided to have their instance translated by tolingo and then the export of the XLIFF was very complicated (or even faulty) because their database was kind-of messed up.
Also - as far as I remember - @dkehne stated, that the only use-case is that all pages of one language in a site have the same source-language. (correct me if I'm wrong) Therefore saving the source-language per page would be duplicated data. 

So this was my attempt to prevent this kind of mess.
Also this is just an explanation of my action, I don't really know what to recommend here.

Reply:
Thanks for your detailed feedback and the good ideas. It took me until now to finish the changes, could you have a look at the updated models and check if they are fine now? Thanks in advance!